texar/torch/modules/pretrained/t5.py

```diff
@@ -0,0 +1,461 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Utils for T5 Modules
+"""
+
+import copy
+import os
+from abc import ABC
+from typing import Any, Set, Dict
+import torch
+
+from texar.torch.modules.pretrained.pretrained_base import PretrainedMixin
+from texar.torch.utils.gin import read_t5_gin_config_file
+
+__all__ = [
+    "PretrainedT5Mixin"
+]
+
+_T5_PATH = "https://storage.googleapis.com/t5-data/pretrained_models/"
+_CHECKPOINT_FILES = {'small': ['checkpoint',
+'model.ckpt-1000000.data-00000-of-00016',
```

Comment:
Let's replace this by a helper method that generates T5 pre-trained file lists, something like this:
```python
def _generate_t5_file_list(ckpt_id: int, ckpt_parts: int) -> List[str]:
    return [
        'checkpoint',
        *[f'model.ckpt-{ckpt_id}.data-{idx:05d}-of-{ckpt_parts:05d}'
          for idx in range(ckpt_parts)],
        f'model.ckpt-{ckpt_id}.index',
        f'model.ckpt-{ckpt_id}.meta',
        'operative_config.gin']
```

Reply:
👍